teeworlds7/user_actions.go

```diff
@@ -52,16 +59,18 @@ func (client *Client) SendPacket(packet *protocol7.Packet) error {
 		// flushPacket.Header.Flags.Compression = true
 
 		flushPacket := client.Session.BuildResponse()
-		client.SendPacket(flushPacket)
+		err = client.SendPacket(flushPacket)
+		if err != nil {
+			return err
+		}
 	}
 
-	for _, queuedChunk := range client.QueuedMessages {
-		// TODO: check if we exceed packet size and only put in as many chunks as we can
-		//       also use a more performant queue implementation then if we unshift it partially
-		//       popping of one element from the queue should not reallocate the entire queued messages slice
-		packet.Messages = append(packet.Messages, queuedChunk)
-	}
-	client.QueuedMessages = nil
+	// TODO: check if we exceed packet size and only put in as many chunks as we can
+	//       also use a more performant queue implementation then if we unshift it partially
+	//       popping of one element from the queue should not reallocate the entire queued messages slice
+	packet.Messages = append(packet.Messages, client.QueuedMessages...)
+
+	client.QueuedMessages = client.QueuedMessages[:0]
```

Comment:
is this better than setting it to nil?

Reply:
I guess it's better to reuse than let append reallocate.